modules/peridynamics/src/materials/ComputeFiniteStrainNOSPD.C

```diff
@@ -131,9 +131,15 @@ ComputeFiniteStrainNOSPD::computeQpStrainRotationIncrements(RankTwoTensor & tota
       const Real p = trFhatinv_1 * trFhatinv_1 / 4.0;
 
       // cos theta_a
-      const Real C1 =
-          std::sqrt(p + 3.0 * Utility::pow<2>(p) * (1.0 - (p + q)) / Utility::pow<2>(p + q) -
-                    2.0 * Utility::pow<3>(p) * (1.0 - (p + q)) / Utility::pow<3>(p + q));
+      const Real C1_squared = p +
+                              3.0 * Utility::pow<2>(p) * (1.0 - (p + q)) / Utility::pow<2>(p + q) -
+                              2.0 * Utility::pow<3>(p) * (1.0 - (p + q)) / Utility::pow<3>(p + q);
+      if (C1_squared <= 0.0)
```

Comment:
It's OK to take the square root of 0. Should this check just be for <0?

Reply:
This is directly from the tensor mechanics module. Since there was an update on the tensor mechanics corresponding class after the initial implementation , so I just updated the peridynamics version to be consistent. For square root of zero, I think it is save to include it in the check.